### PR TITLE
[cosim] support cycle counters (#521)

### PR DESCRIPTION
**Pull request by SharzyL**

### CODE REVIEW

File: ipemu/csrc/vbridge_impl.cc [hunk 1]
```diff
@@ -377,34 +380,44 @@ std::optional<SpikeEvent> VBridgeImpl::spike_step() {
...
       break;
     case PC_SERIALIZE_AFTER:
       break;
     default:
-      FATAL(fmt::format("SpikeStep: invalid pc: {:08X}", pc));
+      FATAL(fmt::format("SpikeStep: invalid new_pc: {:08X}", new_pc));
     }
   }
 
+  // spike does not bump mcycle by itself, so do it manually
```
Comment (Clo91eaf) [inline ipemu/csrc/vbridge_impl.cc:418]: Maybe we should bump `spike_cycles` first and then record it to `mcycle`, or initialize `spike_cycles` with 1.

Comment (SharzyL) [reply ipemu/csrc/vbridge_impl.cc:418]: You mean `mcycle` should start with 1? I did not expect that. May I ask for reasons?

Comment (Clo91eaf) [reply ipemu/csrc/vbridge_impl.cc:418]: sorry, I misunderstand the difference between `spike_cycle` and `mcycle`, you are right.
review passed.

**Review by Clo91eaf — APPROVED** [review]
🥳passed!

File: ipemu/csrc/vbridge_impl.cc [hunk 2]
```diff
@@ -361,12 +361,15 @@ void VBridgeImpl::getCoverage() { return ctx->coveragep()->write(); }
 
 std::optional<SpikeEvent> VBridgeImpl::spike_step() {
   auto state = proc.get_state();
-  reg_t pc = state->pc;
-  auto fetch = proc.get_mmu()->load_insn(pc);
-  //  VLOG(1) << fmt::format("pre-exec (pc={:08X})", state->pc);
+
+  state->mcycle->write((int64_t) get_t() + spike_cycles);
```
Comment (sequencer) [inline ipemu/csrc/vbridge_impl.cc:365]: maybe:
```
  state->mcycle->write((int64_t) get_t() / 10 + spike_cycles);
```
get_t() is the tick, which is should be `cycle / 10`

**Pull request merged**